src/test/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGeneratorTest.java

```diff
@@ -458,4 +460,71 @@ public void testConstructorWithTreeWalkerAuditEvent() {
+        final List<String> expectedTestOne = Arrays.asList(
+                "/CLASS_DEF[./IDENT[@text='InputXpathQueryGeneratorEscapeCharacters']]/"
+                        + "OBJBLOCK/VARIABLE_DEF[./IDENT[@text='testOne']]/ASSIGN/EXPR[./"
+                        + "STRING_LITERAL[@text='&lt;&gt;&apos;&apos;\\&quot;&amp;abc;&amp;lt;"
+                        + "\\u0080\\n']]",
+                "/CLASS_DEF[./IDENT[@text='InputXpathQueryGeneratorEscapeCharacters']]/"
+                        + "OBJBLOCK/VARIABLE_DEF[./IDENT[@text='testOne']]/ASSIGN/EXPR/"
+                        + "STRING_LITERAL[@text='&lt;&gt;&apos;&apos;\\&quot;&amp;abc;&amp;lt;"
+                        + "\\u0080\\n']"
+        );
+        assertEquals(expectedTestOne, actualTestOne,
+                "Generated queries do not match expected ones");
+
+        final int lineNumberTwo = 6;
+        final int columnNumberTwo = 22;
+        final XpathQueryGenerator queryGeneratorTwo = new XpathQueryGenerator(detailAst,
+                lineNumberTwo, columnNumberTwo, testFileText, tabWidth);
+        final List<String> actualTestTwo = queryGeneratorTwo.generate();
+        final List<String> expectedTestTwo = Arrays.asList(
+                "/CLASS_DEF[./IDENT[@text='InputXpathQueryGeneratorEscapeCharacters']]/"
+                        + "OBJBLOCK/VARIABLE_DEF[./IDENT[@text='testTwo']]/ASSIGN/EXPR[./"
+                        + "STRING_LITERAL[@text='&amp;#0;&amp;#X0\\u0001\\r']]",
+                "/CLASS_DEF[./IDENT[@text='InputXpathQueryGeneratorEscapeCharacters']]/"
+                        + "OBJBLOCK/VARIABLE_DEF[./IDENT[@text='testTwo']]/ASSIGN/EXPR/"
+                        + "STRING_LITERAL[@text='&amp;#0;&amp;#X0\\u0001\\r']"
+        );
+        assertEquals(expectedTestTwo, actualTestTwo,
+                "Generated queries do not match expected ones");
+    }
+
+    @Test
+    public void testTextBlocks() throws Exception {
+        final File testFile = new File(getNonCompilablePath(
+                "InputXpathQueryGeneratorTextBlock.java"));
+        final FileText testFileText = new FileText(testFile,
+                StandardCharsets.UTF_8.name());
+        final DetailAST detailAst =
+                JavaParser.parseFile(testFile, JavaParser.Options.WITHOUT_COMMENTS);
+        final int tabWidth = 8;
+
+        final int lineNumberOne = 6;
+        final int columnNumberOne = 25;
+        final XpathQueryGenerator queryGeneratorOne = new XpathQueryGenerator(detailAst,
+                lineNumberOne, columnNumberOne, testFileText, tabWidth);
+        final List<String> actualTestOne = queryGeneratorOne.generate();
+        final List<String> expectedTestOne = Collections.singletonList(
+            "/CLASS_DEF[./IDENT[@text='InputXpathQueryGeneratorTextBlock']]/OBJBLOCK/"
+                    + "VARIABLE_DEF[./IDENT[@text='testOne']]/ASSIGN/EXPR/"
+                    + "TEXT_BLOCK_LITERAL_BEGIN/TEXT_BLOCK_CONTENT"
```

Comment:
do we have ability to check escaped content of text block here  like `TEXT_BLOCK_CONTENT[@text='.......']`?

Reply:
No, I ran IlleagalTokenCheck and the suppressions generated doesn't contain any text attribute.

Reply:
> suppressions generated

please create Xpath manually.

Reply:
If by manual generation you mean by using [XpathQueryGenerator.generate()](https://github.com/checkstyle/checkstyle/blob/2275613f2991e6e5df78e629c3ba379c9bcce6e9/src/main/java/com/puppycrawl/tools/checkstyle/xpath/XpathQueryGenerator.java#L142) then that too doesn't contain text attributes. I was also sceptical about the Xpath generated so had confirm by running a check.

Reply:
> you can reduce text block to minimilistic to keep xpath in sane size

    String testOne = """
        &1line
       >2line
        <3line
        """;


Done

Reply:
please refactor testTextBlocks to not use generator at all.
please make expectedTestOne to have `TEXT_BLOCK_CONTENT[@text='&1line\n>2line\n<3line']` and prove by UT that such xpath suppress violation.

Reply:
moved to https://github.com/checkstyle/checkstyle/issues/9359